src/line_info.c

```diff
@@ -159,7 +163,21 @@ static int print_n_chars(WINDOW *win, const char *s, size_t n, int max_y)
         }
 
         if (win) {
-            wprintw(win, "%c", ch);
+#ifdef HAVE_WIDECHAR
+            // TODO(Jfreegman): Is there a better way to print a single wchar?
+            wchar_t b[sizeof(wchar_t) * 2];
+            wmemcpy(b, &ch, sizeof(ch));
+            b[1] = L'\0';
+            waddwstr(win, b);
```

Comment:
This could use `waddnwstr(WINDOW *win, const wchar_t *wstr, int n);` and skip the intermediate buffer and null termination.